src/main/java/input_feed/distributed/TextToDistributedR1CS.java

```diff
@@ -0,0 +1,156 @@
+
+        // Need at least one constraint per partition!
+        assert(numConstraints >= config.numPartitions());
+        config.beginRuntime("Distribute Constraints");
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationA =
+                distributedCombination(config, filePath + ".a", false);
+
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationB =
+                distributedCombination(config, filePath + ".b", false);
+
+        JavaPairRDD<Long, LinearTerm<FieldT>> combinationC =
+                distributedCombination(config, filePath + ".c", negateCMatrix);
+        config.endRuntime("Distribute Constraints");
+
+        config.beginRuntime("Count combinations");
+
+        config.beginLog("Count A");
+        combinationA.count();
+        config.endLog("Count A");
+
+        config.beginLog("Count B");
+        combinationB.count();
+        config.endLog("Count B");
+
+        config.beginLog("Count C");
+        combinationC.count();
+        config.endLog("Count C");
+
+        config.endRuntime("Count combinations");
+
+        final R1CSConstraintsRDD<FieldT> constraints = new R1CSConstraintsRDD<>(
+                combinationA,
+                combinationB,
+                combinationC,
+                numConstraints
+        );
+
+        return new R1CSRelationRDD<>(constraints, numInputs, numAuxiliary);
+    }
+
+    public Tuple2<Assignment<FieldT>, JavaPairRDD<Long, FieldT>> loadWitness(Configuration config) {
+
+        config.beginRuntime("Distribute Witness");
+        config.beginLog("Distribute witness");
+
+        FieldT field = fieldParameters;
+        String path = filePath;
+        if (useFileSlash) { path = "file://" + path; }
+
+        final int numAux = numAuxiliary;
+        JavaPairRDD<Long, FieldT> auxAssignment =
+                config.sparkContext().textFile(path + ".aux").zipWithIndex().flatMapToPair(
+                        line -> Collections.singleton(new Tuple2<>(line._2, field.construct(line._1))).iterator()
+                ).persist(config.storageLevel());
+
+        JavaPairRDD<Long, FieldT> primaryAssignment =
+                config.sparkContext().textFile(path + ".public").zipWithIndex().flatMapToPair(
+                        line -> Collections.singleton(new Tuple2<>(numAux + line._2, field.construct(line._1))).iterator()
+        ).persist(config.storageLevel());
+
+        JavaPairRDD<Long, FieldT> fullAssignment = auxAssignment.union(primaryAssignment);
+
+        config.beginLog("Count assignment");
+        fullAssignment.count();
+        config.endLog("Count assignment");
+
+        config.endLog("Distribute witness");
+        config.endRuntime("Distribute Witness");
+
+        config.beginLog("Serialize Primary");
+        config.beginRuntime("Serialize Primary");
+        final Assignment<FieldT> primary = new Assignment<>();
+        for (long i=0; i < numInputs; i++){
+            List<FieldT> element = fullAssignment.lookup(i);
```

Comment:
We are now looking up the primary witness value in the beginning of the fullAssignment which is actually the aux witness (see union statement above).

Can we make this work correctly by creating the fullAssignment as `primary u aux` (instead of the other way around) and shifting all linear terms variable indexes accordingly `(index' = index + |primary|` if variable in aux; `index' = index - |aux|` otherwise)